pronotepy/dataClasses.py

```diff
@@ -1991,3 +1993,66 @@ def __init__(self, client: ClientBase, json_dict: dict) -> None:
         self.duration: Optional[datetime.timedelta] = self._resolver(
             lambda v: datetime.timedelta(minutes=int(v)), "duree", strict=False
         )
+
+
+class TeachingSubject(Object):
+    """
+    Represents a subject teached. You shouldn't have to create this class manually.
+
+    Attributes:
+        id (str): id of the subject (used internally)
+        name (str): name of the subject
+        duration (datetime.timedelta): the duration of the subject per week
+        parent_subject_name (Optional[str]): name of the parent subject
+        parent_subject_id (Optional[str]): id of the parent subject (used internally)
+    """
+
+    def __init__(self, json_dict: dict) -> None:
+        super().__init__(json_dict)
+
+        self.id: str = self._resolver(str, "N")
+        self.name: str = self._resolver(str, "L")
+        self._duration: str = self._resolver(str, "volumeHoraire")
+        self.parent_subject_name: str = self._resolver(
+            str, "servicePere", "V", "L", strict=False
+        )
+        self.parent_subject_id: str = self._resolver(
+            str, "servicePere", "V", "N", strict=False
+        )
+
+        if "h" in self._duration:  # duration can be an empty string
```

Comment:
Why can it be an empty string? If the information is not provided, then making `duration` `Optional` would be a better option IMO. If 0 is implied, then this is correct.